example/chatRoom.tsx

```diff
@@ -35,18 +32,17 @@ export const start = () => {
 export const Chat = () => {
   const [message, setMessage] = useState('')
   const [userName, setUserName] = useState('user')
-  const [chatRoomFish, setChatRoomFishName] = useFish(ChatRoomFish, 'lobby')
-  const [chatRoomRegistryFish] = useFish(ChatRoomRegistryFish, 'reg')
+  const [chatRoomName, setChatRoomName] = useState('lobby')
+  const chatRoomFish = useFish(ChatRoomFish.forChannel(chatRoomName))
 
   return (
     <div>
       <div>
         current chat room:{' '}
         <input
-          onChange={({ target }) => setChatRoomFishName(target.value)}
-          value={chatRoomFish ? chatRoomFish.name : ''}
+          onChange={({ target }) => target.value !== chatRoomName && setChatRoomName(target.value)}
```

Comment:
It would be nice to bring this back into the useFish. Probably we can have a check for primitives only or a deepEqual...

Reply:
but most of them hate it to have so many of them in every Component.